sdk/ai/azure-ai-inference/azure/ai/inference/_patch.py

```diff
@@ -73,6 +96,159 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+_P = ParamSpec("_P")
+_R = TypeVar("_R")
+
+
+def _set_attribute(span: AbstractSpan, key: str, value: Any) -> None:
+    if value is not None:
+        span.add_attribute(key, value)
+
+
+def _add_request_chat_message_event(span: AbstractSpan, **kwargs: Any) -> None:
+    for message in kwargs.get("messages", []):
+        try:
+            message = message.as_dict()
+        except AttributeError:
+            pass
+
+        if message.get("role"):
+            name = f"gen_ai.{message.get('role')}.message"
+            span.span_instance.add_event(
+                name=name,
+                attributes={"event.data": json.dumps(message)}
+            )
+
+
+def _add_request_chat_attributes(span: AbstractSpan, **kwargs: Any) -> None:
+    _set_attribute(span, "gen_ai.system", "openai")
```

Comment:
it feels like a `_set_attributes` function that takes an iterable of key/values might be more convenient?